lib/ecto/type.ex

```diff
@@ -637,25 +639,25 @@ defmodule Ecto.Type do
   end
 
   def cast(:time, term) do
```

Comment:
What do you think about keeping the private `cast_time` as before, and move the `pad_usec` logic to the `cast/2` function like in the previous clauses?

```elixir
def cast(:time, term) do
  case cast_time(term) do
    {:ok, time} -> {:ok, zerofy_microsecond(time)}
    :error -> :error
  end
end

def cast(:time_usec, term) do
  case cast_time(term) do
    {:ok, time} -> {:ok, pad_usec(time)}
    :error -> :error
  end
end

defp pad_usec(nil), do: nil
defp pad_usec(%{calendar: _, microsecond: {microsecond, _precision}} = struct) do
  %{struct | microsecond: {microsecond, 6}}
end
```

It would behave similarly to `zerofy_microsecond/1` :)

Reply:
@fertapric you're of course right, done!